src/core/components/Paginate/index.js

```diff
@@ -11,69 +15,59 @@ function makePageNumbers({ start, end }) {
   }
 
   static defaultProps = {
-    perPage: 25,
+    perPage: 20,
     query: {},
-    showPages: 9,
   }
 
   pageCount() {
     return Math.ceil(this.props.count / this.props.perPage);
   }
 
-  visiblePages() {
-    const { currentPage, showPages } = this.props;
-    const pageCount = this.pageCount();
-    const showExtra = Math.floor(showPages / 2);
-    const start = Math.max(1, currentPage - showExtra);
-    const end = Math.min(pageCount, currentPage + showExtra);
-
-    // If we can show all of the pages, show them all.
-    if (pageCount <= showPages) {
-      return makePageNumbers({ start: 1, end: pageCount });
-    // If we are showing less on the right than we should, define the start by the end.
-    } else if (end - currentPage < showExtra) {
-      return makePageNumbers({ start: (end - showPages) + 1, end });
-    // If we are showing less on the left than we should, define the end by the start.
-    } else if (currentPage - start < showExtra) {
-      return makePageNumbers({ start, end: (start + showPages) - 1 });
-    }
-    // We're showing the maximum number of pages on each side, start and end are correct.
-    return makePageNumbers({ start, end });
-  }
-
-  makeLink({ currentPage, page, pathname, query, text }) {
-    let child;
+  makeLink({ currentPage, page, pathname, query, text, className }) {
     if (currentPage === page || page < 1 || page > this.pageCount()) {
-      child = text || page;
+      return (<span key={page} className={
+        classNames('Paginator-item', 'disabled', className )}>
+          {text || page}
+        </span>);
     } else {
       const newQuery = { page, ...query };
-      child = <Link to={{ pathname, query: newQuery }}>{text || page}</Link>;
+      return <Link to={{ pathname, query: newQuery }} className={
+        classNames('Paginator-item', className )}>{text || page}</Link>;
     }
-    return <li key={page} className="paginator--item">{child}</li>;
   }
 
   render() {
     if (this.pageCount() === 1) {
       return null;
     }
-    const { currentPage, pathname, query } = this.props;
+
+    const { currentPage, i18n, pathname, query } = this.props;
+
     return (
-      <ul className="paginator">
-        {this.makeLink({ page: currentPage - 1, currentPage, pathname, query, text: 'Prev' })}
-        {this.visiblePages().map((page) => this.makeLink({ page, currentPage, pathname, query }))}
-        {this.makeLink({ page: currentPage + 1, currentPage, pathname, query, text: 'Next' })}
-      </ul>
+      <div className="Paginator">
+        <div className="Paginator-page-number">
+          {i18n.sprintf(
+            i18n.gettext('Page %(currentPage)s of %(totalPages)s'),
+            { currentPage, totalPages: this.pageCount() }
+          )}
+        </div>
+        <div className="Paginator-links">
+          {this.makeLink({ page: currentPage - 1, currentPage, pathname, query, text: i18n.gettext('Previous'), className: 'Paginator-previous' })}
```

Comment:
I was mostly thinking out loud here.

Since we're creating a new function for the `onClick` each time I don't think react will be able to optimise this as much. Although maybe it just replaces the click handler. If we passed in the function then this might get better optimisation.

Thinking about it more, this should probably just be a link and we pass in the URL. I'm not sure why I didn't do it this way to begin with.

```jsx
<Link
  to={ ...this.url, query: { ...this.url.query, page: currentPage + 1 } }
  className="Paginator-link Paginator-link--next"
  activeClass="Paginator-link--active"
>{i18n.gettext('Next')</Link>
```

I'll file an issue to look at this since it seems unrelated to your changes.